jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/SliderRange.java

```diff
@@ -84,50 +86,53 @@ public void setValue(int thumbIndex, int value) {
     int width = Integer.parseInt(styles[1]);
     String newStyle = setNewStyle(trackStyle, thumbIndex, value, left, width);
 
-    reflectionSetValue(thumbIndex, newThumbStyle, newStyle, trackStyle,value, left, width);
+    reflectionSetValue(thumbIndex, newThumbStyle, newStyle, value, left, width);
 
   }
 
 
-  private boolean isSwitch(String style, int thumbIndex, int value, int left, int width) {
+  private boolean isSwitch(int thumbIndex, int value, int Left, int width) {
+    int localLeft = Left;
+
     switch (thumbIndex) {
-      case 1: return  (value > width + left);
-      case 2: return (value < left);
+      case 1: return  (value > width + localLeft);
+      case 2: return (value < localLeft);
+      default: break;
     }
     return false;
   }
-
-  private String setNewStyle(String style, int thumbIndex, int value, int left, int width) {
-
+  private String setNewStyle(String style, int thumbIndex, int value, int Left, int width) {
+    String localStyle = style;
+    int localLeft = Left;
     switch (thumbIndex) {
       case 1:
-        if (value > left) {
-          if (value > width + left) {
-            left = width + left;
-            width = value - left;
+        if (value > localLeft) {
+          if (value > width + localLeft) {
+            localLeft = width + localLeft;
+            width = value - localLeft;
           } else {
-            width -= value - left;
-            left = value;
+            width -= value - localLeft;
+            localLeft = value;
           }
 
         } else {
-          width += left - value;
-          left = value;
+          width += localLeft - value;
+          localLeft = value;
         }
         break;
       case 2:
-        width += value - width - left;
+        width += value - width - localLeft;
         if (width < 0) {
-          width = left - value;
-          left = value;
+          width = localLeft - value;
+          localLeft = value;
         }
-        break;
+      default: break;
```

Comment:
Do you really need here "default:" Where is break for case 2?

Reply:
case 2 will follow into default and break in there
break to case 2 added.